app/controllers/application_controller.rb

```diff
@@ -31,4 +34,11 @@ def check_entitled(entitlement)
 
     required_entitlements.map { |e| entitlement.send(e) }.all?
   end
+
+  def validate_rbac_groups(user)
+    Thread.new do
+      return unless user.org_admin?
+      RBAC::GroupSeed.new(user).process
```

Comment:
@gmcculloug This can lead to race conditions while we are seeding for the same request we are fetching data from the RBAC service based on those roles. Can this seeding be made synchronous so if there is a failure it is consistent. The seeding shouldn't be taking an inordinate amount of time.

Reply:
Definitely, we should look into doing something like a db lock to protect against a race condition.